doc/ch-test_desc.rst

```diff
@@ -100,6 +104,12 @@ Additional arguments:
   :code:`--dry-run`
     Print summary of what would be tested and then exit.
 
+  :code:`--file-tag TAG`
+    If running an individual Bats file with :code:`--file`, and that file is a
+    :code:`test.bats` to accompany a directory in :code:`examples`,
+    :code:`ch-test` is not smart enough to infer the image tag. Thus, you must
+    specify it with this option.
+
```

Comment:
I had commented earlier about a followup, however, the more I think about it the more I think the `--file-tag` option should be smarter. In our `run` phase we already have some logic that 1) gets a tag from a `CH_TEST_TARDIR` artifact, 2) searches for a corresponding dockerfile, Build script, etc., and 3) modifies the `example` and `image` variables responsible for generating build and run phase tests.

How hard would it be to update this logic to work with a `--file-tag` argument?

I think it would flow much better to specify a tag and phase.

Reply:
The issue I ran into is that a given `test.bats` can have more than one Dockerfile or `Build.*`, so it's not possible to derive a test tag from a `test.bats`. Also, I'm not sure how to map a tag to non-`test.bats` files, e.g. something in `test/run` or `test/build`.

Reply:
That seems right. My thought is that `--file` and `--file-tag` can be used separately or together. For example, if you want to target a file, e.g., `test.bats`, `build_auto.bats`, `run/ch-run_misc.bats`, .etc you would use `--file` tag (don't think a tag needs to be specified with those?) However, if you want to test an example by tag, you could do so with just `--file-tag`. In this case, both `ch-test full --file-tag mpihello`.

Does that cover your issue? The tag is what matters when multiple files share a `test.bats`, not sure about `Build.*` though.